src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,52 +1,136 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 /**
- * <p>Реалізувати свій ArrayList який імплементує інтерфейс List</p>
+ * <p>Реалізувати свій ArrayList який імплементує інтерфейс List.</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private Object[] data;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            this.data = new Object[initialCapacity];
+        } else {
+            throw new IllegalArgumentException("Wrong initial capacity: " + initialCapacity);
+        }
+    }
+
+    public ArrayList() {
+        this.data = new Object[DEFAULT_CAPACITY];
+    }
+
+    private void ensureCapacity(int amount) {
```

Comment:
do we need `int amount` here?

Reply:
I guess yes. How can we check whether we have needed capacity, if we don't know how much is needed capacity?

Reply:
in 1.5